skyportal/handlers/api/observingrun.py

```diff
@@ -0,0 +1,226 @@
+            data['assignments'] = [a.to_dict() for a in data['assignments']]
+
+            # calculate when the targets rise and set
+            for d, a in zip(data['assignments'], run.assignments):
+                d['rise_time_utc'] = a.rise_time.isot
+                d['set_time_utc'] = a.set_time.isot
+
+            return self.success(data=data)
+        runs = ObservingRun.query.all()
+        data = ObservingRunGet.dump(runs, many=True)
+        out = sorted(data, key=lambda d: d['ephemeris']['sunrise_utc'])
+        return self.success(data=out)
+
+    @permissions(['Upload data'])
+    def put(self, run_id):
+        """
+        ---
+        description: Update observing run
+        parameters:
+          - in: path
+            name: run_id
+            required: true
+            schema:
+              type: integer
+        requestBody:
+          content:
+            application/json:
+              schema: ObservingRunPost
+        responses:
+          200:
+            content:
+              application/json:
+                schema: Success
+          400:
+            content:
+              application/json:
+                schema: Error
+        """
+
+        data = self.get_json()
+        run_id = int(run_id)
+        is_superadmin = 'System admin' in [a.id for a in self.current_user.acls]
+
+        orun = ObservingRun.query.get(run_id)
+
+        current_user_id = self.current_user.id if \
+            hasattr(self.current_user, 'username') else self.current_user.created_by_id
+
+        if orun.owner_id != current_user_id and not is_superadmin:
+            return self.error('Only the owner of an observing run can modify '
+                              'the run.')
+        try:
+            new_params = ObservingRunPost.load(data, partial=True)
+        except ValidationError as exc:
+            return self.error('Invalid/missing parameters: '
+                              f'{exc.normalized_messages()}')
+
+        for param in new_params:
+            setattr(orun, param, new_params[param])
+
+        DBSession().add(orun)
+        DBSession().commit()
+        return self.success()
+
+    @permissions(['Upload data'])
+    def delete(self, run_id):
+        """
+        ---
+        description: Delete an observing run
+        parameters:
+          - in: path
+            name: run_id
+            required: true
+            schema:
+              type: integer
+        responses:
+          200:
+            content:
+              application/json:
+                schema: Success
+          400:
+            content:
+              application/json:
+                schema: Error
+        """
+        run_id = int(run_id)
+        is_superadmin = 'System admin' in [a.id for a in self.current_user.acls]
+
+        run = ObservingRun.query.get(run_id)
+
+        current_user_id = self.current_user.id if \
+            hasattr(self.current_user, 'username') else self.current_user.created_by_id
```

Comment:
```suggestion
        current_user_id = self.associated_user_object.id
```
I know there are probably some of these lying around that also need updating, but let's not add more of them in ;)

Reply:
Fixed